stress/src/metrics_latency.rs

```diff
@@ -0,0 +1,229 @@
+    {
+        Self(Arc::new(reader))
+    }
+}
+
+impl MetricReader for SharedReader {
+    fn register_pipeline(&self, pipeline: Weak<Pipeline>) {
+        self.0.register_pipeline(pipeline)
+    }
+
+    fn collect(&self, rm: &mut ResourceMetrics) -> Result<(), MetricError> {
+        self.0.collect(rm)
+    }
+
+    fn force_flush(&self) -> Result<(), MetricError> {
+        self.0.force_flush()
+    }
+
+    fn shutdown(&self) -> Result<(), MetricError> {
+        self.0.shutdown()
+    }
+
+    fn temporality(&self, kind: InstrumentKind) -> Temporality {
+        self.0.temporality(kind)
+    }
+}
+
+const ITERATIONS_PER_THREAD: usize = 10000;
+
+fn main() {
+    let available_threads: usize = std::thread::available_parallelism().map_or(1, |p| p.get());
+
+    measure_update_latency(
+        &format!("half threads = {available_threads}/2, lots of inserts"),
+        available_threads / 2,
+        |_i, j| format!("{}", j % 2000),
+    );
+    measure_update_latency(
+        &format!("half threads = {available_threads}/2, updates only"),
+        available_threads / 2,
+        |_i, _j| String::new(),
+    );
+    measure_update_latency(
+        &format!("twice threads = {available_threads}*2, lots of inserts"),
+        available_threads * 2,
+        |_i, j| format!("{}", j % 2000),
+    );
+    measure_update_latency(
+        &format!("twice threads = {available_threads}*2, updates only"),
+        available_threads * 2,
+        |_i, _j| String::new(),
+    );
+}
+
+fn measure_update_latency(msg: &str, threads_count: usize, keyname: fn(usize, usize) -> String) {
+    let reader = SharedReader::new(
+        ManualReader::builder()
+            .with_temporality(Temporality::Delta)
+            .build(),
+    );
+    let provider = SdkMeterProvider::builder()
+        .with_reader(reader.clone())
+        .build();
+    let histogram = provider.meter("test").u64_counter("hello").build();
+    let mut threads = Vec::new();
+    let mut stats = Vec::new();
+    stats.resize_with(threads_count, || {
+        Arc::new(Mutex::new(HashMap::<u64, u64>::new()))
+    });
+    // run multiple threads and measure how time it takes to update metric
+    for thread_idx in 0..threads_count {
+        let hist = histogram.clone();
+        let stat = stats[thread_idx].clone();
+        threads.push(std::thread::spawn(move || {
+            let mut stat = stat.lock().unwrap();
+            for iter_idx in 0..ITERATIONS_PER_THREAD {
+                let kv = KeyValue::new(keyname(thread_idx, iter_idx), 1);
+                let start = Instant::now();
+                hist.add(1, &[kv]);
+                let curr = stat.entry(start.elapsed().as_nanos() as u64).or_default();
+                *curr += 1;
+            }
+        }));
+    }
+    // collect threads in short intervals
+    let mut total_count = 0;
+    while threads.iter().any(|t| !t.is_finished()) {
+        // collect agressively so we could measure inserts properly,
```

Comment:
trying to understand what this is simulating? Metric collection interval is by default 30 sec or 60 sec in the spec. What is the scenario for collecting every 500 microsec?

Reply:
Basically idea, is to choose collection interval so it wouldn't affect numbers (p99) but still allow to test updates and inserts (for insert I need to clean it regularly, but not too frequent that updates would become inserts).
I updated code with explanation why this number... it could be different, but it should be good enough so it would work in all cases without affecting p99.
( Sorry for rebase,-force push, first commit was a bit rushed... I wasn't happy with, now it's better).